resqpy/olio/relperm.py

```diff
@@ -0,0 +1,316 @@
+            if set(df.columns).issubset({'Sw', 'So', 'Krw', 'Kro', 'Pc'}) and len(set(df.columns)) >= 3:
+               phase_combo = 'water-oil'
+            elif set(df.columns).issubset({'Sg', 'So', 'Krg', 'Kro', 'Pc'}) and len(set(df.columns)) >= 3:
+               phase_combo = 'gas-oil'
+            elif set(df.columns).issubset({'Sg', 'Sw', 'Krg', 'Krw', 'Pc'}) and len(set(df.columns)) >= 3:
+               phase_combo = 'gas-water'
+            else:
+               raise Exception('unexpected number of columns and/or column headers')
+
+         # ensure that missing capillary pressure values are stored as np.nan
+         for col in df.columns:
+            if col.capitalize() == 'Pc':
+               df[col].replace('None', np.nan, inplace = True)
+
+         # convert all values in the dataframe to numeric type
+         df_cols = df.columns
+         df[df_cols] = df[df_cols].apply(pd.to_numeric, errors = 'coerce')
+
+         # ensure that no other column besides Pc has missing values
+         cols_no_pc = [x for x in df.columns if 'pc' != x.lower()]
+         for col in cols_no_pc:
+            if col.capitalize == 'Pc':
+               continue
+            elif (df[col].isnull().sum() > 0) | ('None' in list(df[col])):
+               raise Exception(f'missing values found in {col} column')
+
+         # check that Sw, Kr and Pc values are monotonic and that the Sw and Kr values are within the range 0-1
+         sat_col = [x for x in df.columns if x[0] == 'S'][0]
+         relp_corr_col = [x for x in df.columns if x == 'Kr' + sat_col[-1]][0]
+         relp_opp_col = [x for x in df.columns if (x[0:2] == 'Kr') & (x[-1] != sat_col[-1])][0]
+         assert (df[sat_col].is_monotonic and df[relp_corr_col].is_monotonic and df[
+             relp_opp_col].is_monotonic_decreasing) or \
+             (df[sat_col].is_monotonic_decreasing and df[relp_corr_col].is_monotonic_decreasing and
+                 df[relp_opp_col].is_monotonic), f'{sat_col, relp_corr_col, relp_opp_col} combo is not monotonic'
+         if 'Pc' in df.columns:
+            assert df['Pc'].dropna().is_monotonic or df['Pc'].dropna(
+            ).is_monotonic_decreasing, 'Pc values are not monotonic'
+         for col in ['Sw', 'Sg', 'So', 'Krw', 'Krg', 'Kro']:
+            if col in df.columns:
+               assert df[col].min() >= 0 and df[col].max() <= 1, f'{col} is not within the range 0-1'
+
+      super().__init__(model,
+                       uuid = uuid,
+                       support_root = None,
+                       df = df,
+                       uom_list = uom_list,
+                       realization = realization,
+                       title = title,
+                       column_lookup_uuid = column_lookup_uuid,
+                       uom_lookup_uuid = uom_lookup_uuid)
+      self.phase_combo = phase_combo
+      self.low_sal = low_sal
+      self.table_index = table_index
```

Comment:
I think we need to save these 3 extra attributes as extra metadata items. Then when loading from a uuid, set (and/or check) the values based on that extra metadata.

Reply:
Hi Andy, 
I've gone ahead and saved the attributes as extra_metadata items in the RelPerm class's init method.